middleware/cache.js

```diff
@@ -0,0 +1,36 @@
+const config = require('config')
+const redis = require('redis')
+const { isInCache, formatCacheData, clearOutdatedCache } = require('../services/cache')
+
+const cacheDataClient = redis.createClient(config.redis)
+cacheDataClient.connect()
+
+const saveToCache = async (url, data) => {
```

Comment:
all this logic should be in a service as we discussed. the cacheDataClient should be a part of the service